src/dapps/selectors.ts

```diff
@@ -60,3 +60,17 @@ export const dappCategoriesByIdSelector = createSelector(
 export const dappConnectInfoSelector = (state: RootState) => state.dapps.dappConnectInfo
 
 export const dappFavouritesEnabledSelector = (state: RootState) => state.dapps.dappFavouritesEnabled
+
+export const recentDappsSelector = createSelector(
+  dappsListSelector,
+  recentDappsIdsSelector,
+  (dapps, recentDappIds) => {
+    return recentDappIds.reduce((acc, recentDappId) => {
+      const recentDapp = dapps.find((dapp) => dapp.id === recentDappId)
+      if (recentDapp) {
+        return [...acc, recentDapp]
+      }
+      return acc
+    }, [] as Dapp[])
```

Comment:
nit: I find `reduce` always hard to read compared to a plain old loop :D

Reply:
lol should we actively try to avoid reduce? we haven't been consistent about this, i like reduce but i know you don't 😅

Reply:
lol yeah I think our future selves will thank us for not abusing `reduce`